app/src/main/java/com/jerboa/ui/components/settings/Settings.kt

```diff
@@ -0,0 +1,407 @@
+package com.jerboa.ui.components.settings
+
+import android.graphics.Bitmap
+import android.net.Uri
+import android.util.Log
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.input.KeyboardCapitalization
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.navigation.NavController
+import androidx.navigation.compose.rememberNavController
+import com.jerboa.api.uploadPictrsImage
+import com.jerboa.datatypes.api.SaveUserSettings
+import com.jerboa.db.Account
+import com.jerboa.decodeUriToBitmap
+import com.jerboa.imageInputStreamFromUri
+import com.jerboa.ui.components.common.LargerCircularIcon
+import com.jerboa.ui.components.common.PictrsBannerImage
+import com.jerboa.ui.components.home.SiteViewModel
+import com.jerboa.ui.theme.*
+import kotlinx.coroutines.launch
+
+fun settingsClickWrapper(
+    navController: NavController,
+    account: Account?
+) {
+    account.also {
+        navController.navigate(route = "settings")
+    }
+}
+
+@Composable
+fun SettingsHeader(
+    navController: NavController = rememberNavController(),
+) {
+
+    val backgroundColor = MaterialTheme.colors.primarySurface
+    val contentColor = contentColorFor(backgroundColor)
+
+    TopAppBar(
+        title = {
+            Text(
+                text = "Settings",
+            )
+        },
```

Comment:
Could you move the Save button into this header `actions` ? An example of this is in PostEdit.kt

Reply:
Is this necessary? To do this we will have to store the form data in the SettingsViewModel which will make the implementation very ugly because of the number of items in this form.

Also i was thinking of making this a tabbed page when adding other types of settings, this wouldn't work in that case

Reply:
I wouldn't do tabs, most settings I see in apps have lists for different settings categories, and then those categories have their own activities. I spose its fine to leave as is, but it wouldn't be difficult to pass in the whole form as a single var to the viewmodel.

I'm looking at a few other apps, and the way they seem to do it, which I like a lot, is to not have any save buttons at all, simply do a save immediately after a change. The only field that would be weird for is bio, which we might wanna add a debounce for, but there are already examples of this in the code.

For now you can just put a TODO, to remove the save button.